encoder/vaapiencoder_vp9.cpp

```diff
@@ -159,9 +157,7 @@ YamiStatus VaapiEncoderVP9::doEncode(const SurfacePtr& surface,
 
     PicturePtr picture(new VaapiEncPictureVP9(m_context, surface, timeStamp));
 
-    m_frameCount %= keyFramePeriod();
-    picture->m_type = (m_frameCount ? VAAPI_PICTURE_P : VAAPI_PICTURE_I);
-    m_frameCount++;
+    picture->m_type = (forceKeyFrame ? VAAPI_PICTURE_I : VAAPI_PICTURE_P);
```

Comment:
In the case of VPX there's no need to reset the GOP.  But it is correct that the intraPeriod has to be handled separately from the forceKeyFrame.  The example provided on libyami-utils will force a keyFrame based on intraPeriod but it has to consider the other scenarios and involve other parameters like kf_min_dist kf_max_dist and kf_auto.  I'll submit a better example for this based also on how the driver is supposed to interpret the kf_* variables.